### PR TITLE
add: Implement aa tree in cpp

### PR DESCRIPTION
**Pull request by iattempt**
**Fixes issue:**
#2170 

**Changes:**
1. create aa tree prototype, I will complete it tomorrow.

### CODE REVIEW

File: code/data_structures/aa_tree/aa_tree.cpp [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+    node_type const *find(const_reference value) {
+        
+    }
+
+private:
+};
+
+
+// for test
+#include <iostream>
```
Comment (arnavb) [inline code/data_structures/aa_tree/aa_tree.cpp]: Comment out the test.

Comment (iattempt) [reply code/data_structures/aa_tree/aa_tree.cpp]: not complete~  😅

**Comment by iattempt** [comment]
@AdiChat
thanks for uniform code style

but, I'm sorry for scope of code style, some type or other need ordered definition, it is difficult to implement.
then, I use a viable alternative: still use public->private->protected, but may have multi times, like:
```
public:
  ...
protected:
  ...
public:
  ...
private:
  ...
```

could we use this way or other recommend?

**Comment by arnavb** [comment]
Sure, sure, Just make sure that the main public interface comes first.

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**